File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.swift

import Foundation

func solution(_ s:String) -> Int {
  var text = ""
  var answer = ""
  let numDict = ["zero" : "0", "one" : "1", "two" : "2",
                 "three" : "3", "four": "4", "five" : "5",
                 "six" : "6", "seven" : "7", "eight" : "8", "nine" : "9"]
  for i in s {
    if !i.isNumber {
      text += String(i)
    } else {
      answer += String(i)
    }
    if numDict.contains(where: { $0.key == text }) {
      answer += String(numDict[text]!)
      text = ""
    }
  }
  
  return Int(answer)!
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    dict = {
        'zero': '0',
        'one': '1',
        'two': '2',
        'three': '3',
        'four': '4',
        'five': '5',
        'six': '6',
        'seven': '7',
        'eight': '8',
        'nine': '9'
    }
    
    str = ''
    answer = ''
    for ss in s:
        if ss.isdecimal() == False:
            str += ss
        else:
            if str in dict:
                answer = answer + dict[str] + ss
                str = ''
            else:
                answer = answer + ss
                str = ''
        if str in dict:
            answer += dict[str]
            str = ''
    return int(answer)